### PR TITLE
[WIP] protos: Remove unused imports

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: protos: Remove unused imports
Additional Description:

not sure if these are safe to remove, but im noticing that unused (?) imports in proto files are emitting lots of warning across CI builds

one doubt i have is that some of the "unused" imports are `versioning` and `annotations` and im wondering if these need to be brought into the namespace for some reason to do with validation or custom compilation

i have just removed a few to see whether it had any effect in CI, i would be happy to remove the rest if this is wanted

Risk Level: ?
Testing: 
Docs Changes:
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by htuch** [comment]
@phlax makes sense to me. I wonder if you can add a CI validation in the `api` job that verifies we don't have any of these warnings when you are done to prevent regression.

File: api/envoy/config/bootstrap/v3/bootstrap.proto [hunk 1]
```diff
@@ -22,7 +22,6 @@ import "google/protobuf/wrappers.proto";
 
 import "udpa/core/v1/resource_locator.proto";
 
-import "envoy/annotations/deprecation.proto";
```
Comment (phlax) [inline api/envoy/config/bootstrap/v3/bootstrap.proto:25]: removing this triggers an error

```
ERROR: /build/tmp/_bazel_bazel/b570b5ccd0454dc9af9f65ab1833764d/external/envoy_api/envoy/config/bootstrap/v3/BUILD:7:18: Generating Descriptor Set proto_library @envoy_api//envoy/config/bootstrap/v3:pkg failed (Exit 1) protoc failed: error executing command bazel-out/host/bin/external/com_google_protobuf/protoc '--proto_path=bazel-out/k8-fastbuild/bin/external/com_google_protobuf/_virtual_imports/any_proto' ... (remaining 178 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
envoy/config/bootstrap/v3/bootstrap.proto:276:27: Option "(envoy.annotations.disallowed_by_default)" unknown. Ensure that your proto definition file imports the proto which defines the option.
```

not sure if this is my bad...

Comment (htuch) [reply api/envoy/config/bootstrap/v3/bootstrap.proto]: Yeah, you need that one. Was it generating a warning before?

Comment (phlax) [reply api/envoy/config/bootstrap/v3/bootstrap.proto]: i think so - but i need to check. it seems like there are others that are also required - which is making it quite slow to go through.

Ill update as i have time...

**Comment by phlax** [comment]
closed in favour of #13401

**Pull request closed without merging**